internal/utils/namespaces.go

```diff
@@ -18,6 +19,16 @@ import (
 // If the namespace is not yet available a list of the components being waited
 // on will be provided.
 func IsNamespaceAvailable(ctx context.Context, cluster clusters.Cluster, namespace string) (waitForObjects []runtime.Object, available bool, err error) {
+	// if the namespace itself isn't available yet, not ready
+	_, err = cluster.Client().CoreV1().Namespaces().Get(ctx, namespace, metav1.GetOptions{})
+	if err != nil {
+		if errors.IsNotFound(err) {
+			waitForObjects = append(waitForObjects, &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: namespace}})
+			err = nil
+		}
+		return
+	}
+
```

Comment:
This is just a short-circuit which allows code following it to assume namespace availability, and simply makes the readiness check faster when executed before the namespace is available.